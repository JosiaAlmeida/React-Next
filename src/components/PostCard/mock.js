export const PostCardMock = {
    id: 1,
    title: 'title 1',
    body:'body 1',
    cover: 'img.png'
}