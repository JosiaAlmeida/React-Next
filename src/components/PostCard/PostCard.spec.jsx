import { render, screen } from "@testing-library/react"
import {PostCard} from '.'
import {PostCardMock} from './mock'

const Props = PostCardMock
describe('<PostCard />',()=>{
    it('should render PostCard correctly', ()=>{
        render(<PostCard {...Props} /> )
        expect(screen.getByRole('img', {name: /title 1/i})).toHaveAttribute('src', Props.cover)
        expect(screen.getByRole('heading', {name: /title 1/i})).toBeInTheDocument()//Pegando cabeÃ§alho
        expect(screen.getByText('body 1')).toBeInTheDocument() //Pegando textos
    })
    it('should match snapshot', ()=>{
        const {container} = render(<PostCard {...Props} /> )
        expect(container.firstChild).toMatchSnapshot()
    })
})