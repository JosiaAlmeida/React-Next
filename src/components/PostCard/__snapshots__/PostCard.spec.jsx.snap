// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <div
    class="post-content"
  >
    <img
      alt="title 1"
      src="img.png"
    />
    <h1>
       
      title 1
       - 
      1
       
    </h1>
    <p>
       
      body 1
       
    </p>
  </div>
</div>
`;
